### PR TITLE
Generate prow config from simpler config file

### LINKED ISSUE #1471: Prow job config should be generated from a simpler config
from slack: 

would it be a bad idea to have a job config generation for the prow jobs? Is anyone else doing this? It seems like most of our jobs could be greatly simplified with a config like
  ```- name: integ-istioctl-k8s-presubmit-tests
    command: [prow/integ-suite-kind.sh, test.integration.istioctl.kube.presubmit]
    requirements: [kind]```
Rather than ~20 lines of config. It could be something very opinionated, such as adding things like `if requirements == boskos, set max_concurrency=5`. Some benefits to this:
• All options are standardized. We won't have people accidentally configuring tests slightly different in ways they don't understand. I think there are a number of tests that are misconfigured (extra permissions they don't need, always_run = false when they didn't mean it, inconsistent naming, etc)
• We also have a ton of duplication across branches that could be simplified
• We can get rid of anchors which is not recommended
• Single place to config something. Instead of having the builder image across 50 different jobs we can configure it in one place. Instead of specifying details like resource requests for each job, we could have a couple profiles maybe (default, highmem, highcpu, ...)

### PR DESCRIPTION
**Pull request by howardjohn**
Fixes https://github.com/istio/test-infra/issues/1471

I may have messed up the vendor code, it got a little funky pulling in the new test-infra repo. I will double check this part.

This PR adds the code to do the transformation and creates a file using the new job syntax for istio/istio master. The generated config is virtually identical, with the following differences:

(with context field ignored - will be removed soon in https://github.com/istio/test-infra/issues/1460):
```
Presubmit diff:

Diff for release-test-master
JobBase.MaxConcurrency: 0 != 5


Postsubmit diff:

Diff for e2e-mixer-no_auth-master
JobBase.MaxConcurrency: 0 != 5

Diff for istio-pilot-e2e-envoyv2-v1alpha3-master
JobBase.MaxConcurrency: 0 != 5

Diff for e2e-dashboard-master
JobBase.MaxConcurrency: 0 != 5

Diff for e2e-bookInfoTests-envoyv2-v1alpha3-master
JobBase.MaxConcurrency: 0 != 5

Diff for e2e-simpleTests-master
JobBase.MaxConcurrency: 0 != 5

Diff for istio-kind-simpleTest-master
JobBase.Spec.Containers[0].Resources.Requests["memory"].i.value: 2147483648 != 3221225472
JobBase.Spec.Containers[0].Resources.Requests["memory"].s: "2Gi" != "3Gi"

Diff for istio-pilot-e2e-envoyv2-v1alpha3-k8s-latest-master
JobBase.MaxConcurrency: 0 != 5
```

So basically just some post submit tests are missing a maxConcurrency field the current config

In the future I will add a presubmit gate to ensure the config is generated (and check in the new generated istio/istio config - it is semantically the same but syntactical differences)

### CODE REVIEW

**Comment by howardjohn** [comment]
The vendor update is to pull in an updated kubernetes/test-infra. I'll break that into a different PR once this is ready to merge

**Comment by howardjohn** [comment]
@chases2

**Comment by howardjohn** [comment]
@utka can you take a look?

**Comment by utka** [comment]
The tests are still failing

**Comment by howardjohn** [comment]
@utka I'll make a separate PR to update the vendor before checking this is, I was hoping you could look at the new config before then

**Comment by utka** [comment]
Sorry, which config? there is more then 2k files in this PR :)

**Comment by howardjohn** [comment]
@utka I created https://github.com/istio/test-infra/pull/1497 to split the change to what actually needs review, please take a look at that. Should be 3 files only
